cql3/functions/functions.hh

```diff
@@ -16,6 +16,7 @@
 #include "cql3/functions/function_name.hh"
 #include "schema/schema.hh"
 #include <unordered_map>
+#include "data_dictionary/user_types_metadata.hh"
```

Comment:
> Once we create types atomically the code which is before commit
> may depend on newly added types, so it has to access both old and
> new types. New storage called in_progress_types_storage was added.

Would it be simpler to sort the dependencies and run each change separately?

Reply:
We thought about this at the beginning but it has one problem, nobody knew what to do with cyclic dependencies (if there are any now, or in the future), or even what should be the order of execution.

Also second reason is that this way should allow to merge command application and snapshot application code paths. As snapshot application will become exactly like command but with more mutations applied at once.

And third reason for me is that it's easier to reason about when there is a clear interface to deal with, sorting dependencies promotes hidden dependencies and when no order is guaranteed then code must be more explicit and hopefully more readable.

Reply:
I think I explained reasoning behind current design, we can resolve it